### ISSUE TITLE
`clippy::incompatible-msrv` fires even when the code is configured to only show up in the correct Rust version with `rustversion`

### ISSUE THREAD

**Issue by JSorngard** [rust-lang/rust-clippy#16450]
### Summary

If I have a piece of code that uses Rust features that were only added in version X, but I use the [`rustversion`](https://crates.io/crates/rustversion) crate to config that code away unless the compiler is of version X or newer, `clippy::incompatible-msrv` still fires even though (I think) it shouldn't.

### Lint Name

incompatible-msrv

### Reproducer

I tried this code:

```rust
#[rustversion::since(1.83.0)]
macro_rules! foo {
    ($tpe:ty) => {
        pub const fn bar(x: Option<$tpe>) -> $tpe {
            x.expect("this was stabilized in const in 1.83.0")
        }
    };
}

macro_rules! do_the_thing {
    ($tpe:ty) => {
        #[rustversion::since(1.83.0)]
        foo! {$tpe}

        // Other functions active on all Rust-versions
        // ...
    };
}

do_the_thing! {i8}
```
With a `Cargo.toml` file with the field `rust-version="1.56"` added to the `[package]` heading.

I saw this output:

```text
> cargo clippy
warning: current MSRV (Minimum Supported Rust Version) is `1.56.0` but this item is stable in a `const` context since `1.83.0`
  --> src\lib.rs:5:15
   |
 5 |             x.expect("this was stabilized in const in 1.83.0")
   |               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
...
20 | do_the_thing! {i8}
   | ------------------ in this macro invocation
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/rust-1.93.0/index.html#incompatible_msrv
   = note: `#[warn(clippy::incompatible_msrv)]` on by default
   = note: this warning originates in the macro `foo` which comes from the expansion of the macro `do_the_thing` (in Nightly builds, run with -Z macro-backtrace for more info)

warning: `msrv_oddities` (lib) generated 1 warning
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.02s
```

I expected to see this happen:

No lints should trigger, since no code that needs a Rust version of 1.83 or later is compiled when the active compiler has a version below 1.83. This can also be seen by running  `cargo msrv verify` from the `cargo-msrv` library.

This code has worked previously with no issues. I discovered it when CI for a PR for an unrelated changed failed: https://github.com/JSorngard/compile_time_sort/actions/runs/21316602710/job/61360238171?pr=57

### Version

```text
rustc 1.93.0 (254b59607 2026-01-19)
binary: rustc
commit-hash: 254b59607d4417e9dffbc307138ae5c86280fe4c
commit-date: 2026-01-19
host: x86_64-pc-windows-msvc
release: 1.93.0
LLVM version: 21.1.8
```

**Comment by Jarcho** [comment]
`rustversion` does expand to `#[allow(clippy::incompatible_msrv)]`, but allow attributes don't work when applied to macro expansions. This is not going to be fixed here since we won't parse `rustversion` attributes and we can't do anything about the `allow` attribute not working.

**Comment by JSorngard** [comment]
I see. This makes it sound as if it should never have worked, but it has worked for more than a year. What changed?

**Comment by Jarcho** [comment]
The lint changed to check the MSRV for const contexts correctly in `1.90`. Before that it always checked the non-const stabilization version.

**Comment by JSorngard** [comment]
I see. Thanks!
